File: frontend-admin/src/api/extraction.ts
import { http } from './http'

export interface ExtractionDetail {
  id: number
  contractId: number | null
  applicationNo: string
  applicationStatus: string
  applicationTitle: string | null
  applicationType: string | null
  applyDate: string | null
  supplierName: string | null
  itemName: string | null
  itemModel: string | null
  unit: string | null
  quantity: string | null
  purchaseUnitPrice: string | null
  purchaseTotalAmount: string | null
  currency: string | null
  expectedDeliveryDate: string | null
  deliveryLocation: string | null
  paymentTerms: string | null
  message: string | null
  createTime: string | null
  extractedAt: string | null
  sourceJson: string | null
  resultJson: string | null
}

export interface ExtractionListItem {
  id: number
  applicationNo: string
  applicationTitle: string | null
  applicationStatus: string
  supplierName: string | null
  itemName: string | null
  applicationType: string | null
  createTime: string | null
  extractedAt: string | null
}

export interface ExtractionListResponse {
  total: number
  items: ExtractionListItem[]
}

export const extractionApi = {
  list(params: { page?: number; size?: number; status?: string }) {
    return http.get<unknown, { code: number; message: string; data: ExtractionListResponse }>(
      '/extractions',
      { params }
    )
  },
  get(id: number) {
    return http.get<unknown, { code: number; message: string; data: ExtractionDetail }>(
      `/extractions/${id}`
    )
  },
  confirm(id: number) {
    return http.patch<unknown, { code: number; message: string; data: null }>(
      `/extractions/${id}/confirm`
    )
  }
}

File: backend/src/main/java/com/example/contractagent/extraction/Extraction.java
package com.example.contractagent.extraction;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@TableName("extraction")
public class Extraction {
    @TableId(type = IdType.AUTO)
    private Long id;
    private Long contractId;
    private String supplierName;
    private String itemName;
    private String itemModel;
    private String unit;
    private BigDecimal quantity;
    private BigDecimal purchaseUnitPrice;
    private BigDecimal purchaseTotalAmount;
    private LocalDate expectedDeliveryDate;
    private String shippingMethod;
    private String paymentTerms;
    private String deliveryLocation;
    private String applicationNo;
    private String applicationTitle;
    private LocalDate applyDate;
    private String applicationType;
    private String applicantName;
    private String currency;
    private String taxRateName;
    private String applicationStatus;
    private LocalDateTime createTime;
    private String message;
    private BigDecimal confidence;
    private String rawQuote;
    private LocalDateTime extractedAt;
    private String sourceJson;
    private String resultJson;
    private String sourceType;
    private String sourceEventId;
    private String sourceSenderId;
}
